lib/dialects/mariadb/query.js

```diff
@@ -179,19 +179,21 @@ class Query extends AbstractQuery {
     if (!this.model || !this.model.fieldRawAttributesMap) {
       return;
     }
+    // Occurs when using mariadb driver with mysql
+    const wasParsed = rows.meta.some(m => m.type === 'JSON');
     for (const _field of Object.keys(this.model.fieldRawAttributesMap)) {
       const modelField = this.model.fieldRawAttributesMap[_field];
-      if (modelField.type instanceof DataTypes.JSON) {
+      const { type, fieldName } = modelField;
+      if (type instanceof DataTypes.JSON) {
         //value is return as String, no JSON
-        rows = rows.map(row => {
-          row[modelField.fieldName] = row[modelField.fieldName] ? JSON.parse(
-            row[modelField.fieldName]) : null;
+        for (const r of rows) {
+          if (!wasParsed) {
+            r[fieldName] = r[fieldName] ? JSON.parse(r[fieldName]) : null;
+          }
           if (DataTypes.JSON.parse) {
-            return DataTypes.JSON.parse(modelField, this.sequelize.options,
-              row[modelField.fieldName]);
+            DataTypes.JSON.parse(modelField, this.sequelize.options, r[fieldName]);
```

Comment:
I'm afraid this is still not equivalent to the original code. Now you're computing something but not saving it anywhere. It seems to me that using `rows = rows.map(/* ... */)` is probably the best

Reply:
Let me re-read this entire segment...

Reply:
This doesn't make sense since `rows` is never returned only new arrays are spawned of it and then iterated.
Why should a full row be replaced with the result of a single column being parsed? This logic does not make any sense to me and seems to be in error.

Reply:
@SimonSchick Ah, you're very right, I was judging your code assuming the current code was somewhat correct, but as you said the `rows` variable is reassigned but unused after that point!

Sorry about that. Well, since the current code is incorrect, I wonder what it should be doing instead...

Reply:
Do you think this weird part of the code (in which the element is reassigned to no use) can be the cause for the issue you're trying to solve? And if yes, why would it happen only when "mixing" mysql and mariadb? Or did we just stumble on a different issue?

Reply:
No, the issue is that the data is already parsed and the JSON.parse ends up throwing :)